csr/csr.go

```diff
@@ -72,7 +73,7 @@ func VerifyCSR(ctx context.Context, csr *x509.CertificateRequest, maxNames int,
 	names := NamesFromCSR(csr)
```

Comment:
Idea (not for this PR but for a future PR): Add a csr.IdentsFromCSR which does exactly what it says on the tin, then _reimplement NamesFromCSR as a wrapper around that_ which just drops all the identifier metadata on the floor so we don't have to have the logic twice. Then the places (like here) which want identifiers can use IdentsFromCSR, and the places (like the CA, for now) which don't can still use NamesFromCSR.